### ISSUE TITLE
LF 1.15

### ISSUE THREAD

**Issue by sofiafaro-da** [digital-asset/daml#12051]
Previous LF version ticket: https://github.com/digital-asset/daml/issues/9914

Issue to keep track of things to do before releasing LF 1.15. This version includes interfaces.

**Comment by stefanobaghino-da** [comment]
@moisesackerman-da @akrmn What else is part of this apart from #14890?

**Comment by dylant-da** [comment]
> @moisesackerman-da @akrmn What else is part of this apart from #14890?

https://github.com/digital-asset/daml/issues/14833

**Comment by akrmn** [comment]
Thanks @dylant-da. 

@stefanobaghino-da we also have ~a few~ 25 remaining TODOs/FIXMEs in the repo (some of them actually already done), @dylant-da and I are currently going through them.

```bash
> git grep -n "/12051"
compiler/damlc/daml-visual/src/DA/Daml/Visual.hs:143:      -- TODO https://github.com/digital-asset/daml/issues/12051
compiler/damlc/daml-visual/src/DA/Daml/Visual.hs:148:      -- TODO https://github.com/digital-asset/daml/issues/12051
compiler/damlc/daml-visual/src/DA/Daml/Visual.hs:274:    -- TODO https://github.com/digital-asset/daml/issues/12051
compiler/damlc/tests/BUILD.bazel:89:        # TODO https://github.com/digital-asset/daml/issues/12051
compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs:37:  --           https://github.com/digital-asset/daml/issues/12051
compiler/damlc/tests/src/DamlcTest.hs:32:    -- TODO https://github.com/digital-asset/daml/issues/12051
compiler/damlc/tests/src/DamlcTest.hs:39:-- TODO https://github.com/digital-asset/daml/issues/12051
compiler/damlc/tests/src/DamlcTest.hs:164:-- TODO https://github.com/digital-asset/daml/issues/12051
compiler/damlc/tests/src/DamlcTest.hs:273:--              -- TODO https://github.com/digital-asset/daml/issues/12051
compiler/damlc/tests/src/DamlcTest.hs:278:--              -- TODO https://github.com/digital-asset/daml/issues/12051
daml-lf/encoder/BUILD.bazel:119:                # TODO https://github.com/digital-asset/daml/issues/12051
daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/CommandPreprocessor.scala:55:  // TODO: https://github.com/digital-asset/daml/issues/12051
daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/PackageInterfaceSpec.scala:25:  // TODO https://github.com/digital-asset/daml/issues/12051
daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MetaDataTest.scala:130:    // TODO https://github.com/digital-asset/daml/issues/12051:
daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:475:  // TODO https://github.com/digital-asset/daml/issues/12051
daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/PackageInterface.scala:290:  // TODO: https://github.com/digital-asset/daml/issues/12051
daml-lf/language/src/test/scala/com/digitalasset/daml/lf/language/AstSpec.scala:16:// TODO https://github.com/digital-asset/daml/issues/12051
daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/ReplayCommand.scala:23:  // TODO: https://github.com/digital-asset/daml/issues/12051
daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/CollisionSpec.scala:18:  // TODO https://github.com/digital-asset/daml/issues/12051
docs/BUILD.bazel:640:    # FIXME: https://github.com/digital-asset/daml/issues/12051
language-support/java/codegen/BUILD.bazel:180:    # TODO https://github.com/digital-asset/daml/issues/12051
language-support/java/codegen/BUILD.bazel:203:    # TODO https://github.com/digital-asset/daml/issues/12051
language-support/java/codegen/BUILD.bazel:330:    # TODO https://github.com/digital-asset/daml/issues/12051
language-support/java/codegen/BUILD.bazel:343:    # FIXME: https://github.com/digital-asset/daml/issues/12051
ledger/participant-integration-api/src/main/scala/platform/store/ChoiceCoder.scala:8:// TODO: https://github.com/digital-asset/daml/issues/12051
```

**Comment by S11001001** [comment]
Fixed by #15614.
